Main.elm

```diff
@@ -14,52 +15,67 @@ import Json.Encode
 type Msg
     = NewEmail String
     | NewPassphrase String
-    | NewData String
+    | NewData (Maybe String)
+    | UpdateContent String
     | NewError String
     | GetData
     | Lock
-    | SetData String
     | DataSaved String
     | DataNotSaved String
-    | UnStored Time.Time
+    | Modifying Time.Time
     | BlurSelection
     | CopySelection
     | ToggleReveal
+    | LastModified Time.Time
 
 
 type alias Model =
     { lock : Bool
     , email : String
     , passphrase : String
     , content : String
-    , stored : Bool
+    , loadedContent :
+        -- may be desynchronized with "content", only used to redraw the
+        -- contentEditable with new decrypted content
+        String
+    , modified : Bool
     , error : String
     , reveal : Bool
+    , lastModified : Maybe Time.Time
     }
 
 
 init : ( Model, Cmd msg )
 init =
-    Model True "" "" "" False "" False ! []
-
-
-
--- Handle contenteditable events
-
-
-innerHtmlDecoder =
-    Json.Decode.at [ "target", "innerHtml" ] Json.Decode.string
+    Model True "" "" "" "" False "" False Nothing ! []
 
 
 
 -- Update
 
 
-update : Msg -> Model -> ( Model, Cmd msg )
+update : Msg -> Model -> ( Model, Cmd Msg )
 update message model =
     case message of
-        UnStored time ->
-            { model | stored = False } ! []
+        Modifying now ->
+            -- The content is being edited, should we save? (aka debounce)
+            let
+                lastModified =
+                    Maybe.withDefault 0 model.lastModified
+
+                elapsed =
+                    Debug.log "Elapsed" <|
+                        now
```

Comment:
The indentation seems a bit strange to me

`Debug.log "Elapsed" <| now - lastModified` ?